lms/djangoapps/program_enrollments/api/v1/views.py

```diff
@@ -124,3 +133,222 @@ def get(self, request, program_key=None):
+                    self._course_run = course_run
+                    self._course_key = course_key
+                    return
+        raise Http404()
+
+
+class ProgramCourseEnrollmentsView(ProgramCourseRunSpecificViewMixin, APIView):
+    """
+    A view for enrolling students in a course through a program,
+    modifying program course enrollments, and listing program course
+    enrollments
+
+    Path: /api/v1/programs/{program_uuid}/courses/{course_id}/enrollments
+
+    Accepts: [POST]
+
+    ------------------------------------------------------------------------------------
+    POST
+    ------------------------------------------------------------------------------------
+
+    Returns:
+     * 200: Returns a map of students and their enrollment status.
+     * 207: Not all students enrolled. Returns resulting enrollment status.
+     * 401: User is not authenticated
+     * 403: User lacks read access organization of specified program.
+     * 404: Program does not exist, or course does not exist in program
+     * 422: Invalid request, unable to enroll students.
+    """
+    authentication_classes = (
+        JwtAuthentication,
+        OAuth2AuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    permission_classes = (permissions.JWT_RESTRICTED_APPLICATION_OR_USER_ACCESS,)
+    pagination_class = ProgramEnrollmentPagination
+
+    def post(self, request, program_uuid=None, course_id=None):
+        """
+        Enroll a list of students in a course in a program
+        """
+        self.check_course_existence_and_membership()
+        results = {}
+        seen_student_keys = set()
+        enrollments = []
+
+        if not isinstance(request.data, list):
+            raise ValidationError("invalid enrollment record")
+        if len(request.data) > MAX_ENROLLMENT_RECORDS:
+            return Response(
+                'enrollment limit 25', status.HTTP_413_REQUEST_ENTITY_TOO_LARGE
+            )
+
+        try:
+            for enrollment_request in request.data:
+                error_status = self.check_enrollment_request(enrollment_request, seen_student_keys)
+                if error_status:
+                    results[enrollment_request["student_key"]] = error_status
+                else:
+                    enrollments.append(enrollment_request)
+        except (KeyError, ValidationError, TypeError):
+            return Response('invalid enrollment record', status.HTTP_422_UNPROCESSABLE_ENTITY)
+
+        program_enrollments = self.get_existing_program_enrollments(program_uuid, enrollments)
+        for enrollment in enrollments:
+            student_key = enrollment["student_key"]
+            if student_key in results and results[student_key] == CourseEnrollmentResponseStatuses.DUPLICATED:
+                continue
+            results[student_key] = self.enroll_learner_in_course(enrollment, program_enrollments)
```

Comment:
I feel like this line could benefit from the `student_key in results` check as well. Why not setup a condition to handle the missing student_key case?

Reply:
well, this kind of is that case, as of this point we've only added error statuses to the results.
Maybe I should add a case checking for student_key in results and logging it, because that would be unexpected

Reply:
I wouldn't. IMHO, if you have already verified that these are good entries, the logging would just be source code noise.